src/structure.js

```diff
@@ -487,21 +492,31 @@ function hasIn(cursor, path) {
   return cursor.getIn(path, NOT_SET) !== NOT_SET;
 }
 
-function pathString(path) {
-  var topLevel = 'global';
-  if (!path || !path.length) return topLevel;
-  return [topLevel].concat(path).join('|');
+function listListenerMatching (listeners, path) {
+  var pathArray = [],
+      matches = [];
+
+  path = path || [];
+
+  path.forEach(function(pathPart) {
+    pathArray = pathArray.concat(pathPart);
```

Comment:
The same problem applies to the original code the only difference is that there is an implied toString call placed on all parts concat'd together in the original code because of the .join, I could simply call toString on each pathPart but for plain JS objects you will get the generic '[object Object]' string and I'm not sure what you get for the other supported keys.